tests/system/marathon_common_tests.py

```diff
@@ -1268,3 +1271,167 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    common.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_regions(app, region)
+    common.assert_app_in_zones(app, zones)
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+@common.require(len(get_biggest_cluster_region()[1]) > 1, "Cluster must have at least 2 zones in the biggest region")
+def test_faultdomains_region_and_zone():
+    """Tests if the applications with a `region` and a `zone` defined are always starting on the
+       same region and zone
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+    single_zone = zones[0]
+
+    # Create an app pinned on the given region and zone.
+    #
+    # We are starting more instances than the number of zones in the region in order to increase
+    # the chance of a faulty instance to appear on a different zone.
+    #
+    app_def = apps.faultdomain_app(
+        region=region,
+        zone=single_zone,
+        instances=len(zones) * 2,
+        app_id="region-and-zone")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    common.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region, and on *all* the region zones
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_regions(app, region)
+    common.assert_app_in_zones(app, single_zone)
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+@common.require(len(get_all_cluster_regions()) > 1, "Cluster must have at least 2 regions")
+def test_faultdomains_region_unique():
+    """Tests if the applications with a ["@region", "UNIQUE"] constraint correctly starts the apps
+       across all regions in the cluster
+    """
+
+    # Find out all the regions in the cluster
+    regions = get_all_cluster_regions().keys()
+
+    # Create an app that should be deployed on a unique region
+    # The number of instances should be the number of regions available in the system
+    app_def = apps.faultdomain_app(
+        instances=len(regions),
+        constraints=[["@region", "UNIQUE"]],
+        app_id="region-unique"
+    )
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    common.deployment_wait(app_id=app_def['id'])
+
+    # Check if the application was launched correctly
+    app = client.get_app(app_def['id'])
+    (used_regions, used_zones) = get_used_regions_and_zones(get_app_domains(app))
+
+    assert_that(used_regions, equal_to(regions), "App {} was not launched in all regions: {} instead of {}".format(
```

Comment:
Isn't this the same as `common.assert_app_in_regions`?

Reply:
Indeed :)